setup/README.md

```diff
@@ -0,0 +1,6 @@
+# Instructions
+
+1. The clang-format binary is what styles your code automatically. It needs to placed somewhere in your path. I place it in ~/.local. I am running a Ubuntu system. For other systems maybe it's best to look up online how to install clang-format.
```

Comment:
There's a package "clang-format" in apt-get. Any reason we should add the executable in our repository and not just refer to it? Shipping an executable implies a certain environment to run (not very universal) and it requires their License file.

Reply:
I haven't tried it by installing. I think I initially tried to install 3.4, which is no longer in the apt cache, so I figured it just doesn't exist as a package in Ubuntu 16.04. . I didn't realize there were newer versions in the cache until later. Can you try installing and seeing if it works? I will update accordingly.

Reply:
There are multiple versions (clang-format-3.5 up to clang-format-5.0). I installed the default clang-format. Might be the reason my git hook doesn't work.